File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/backend/src/main/java/it/unical/inf/ea/backend/dto/TransactionDTO.java
package it.unical.inf.ea.backend.dto;

import it.unical.inf.ea.backend.dto.enums.TransactionState;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Past;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
public class TransactionDTO {

    @NotNull
    private String id;

    @Past
    private LocalDateTime creationTime;

    @Min(0)
    @NotNull
    private BigDecimal amount;

    @NotNull
    private TransactionState transactionState;

    @NotNull
    private String paymentMethod;

    @NotNull
    private String paymentMethodOwner;

}

File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/frontend/app/src/main/java/com/android/frontend/dto/TransactionDTO.kt
package com.android.frontend.dto

import java.math.BigDecimal
import java.time.LocalDateTime

data class TransactionDTO(
    var id: String,
    var creationTime: LocalDateTime,
    var amount: BigDecimal,
    var transactionState: TransactionState,
    var paymentMethod: String,
    var paymentMethodOwner: String
){
    enum class TransactionState {
        REJECTED,
        COMPLETED
    }
}
